File: c/Numbers/factorial.c
/**
 * calculate the factorial of a positive integer
 *
 */
#include <stdio.h>
#include <stdlib.h>

long factorial(int number)
{
	if (number == 0)
		return 1;
	else if (number == 1)
		return 1;
	else
		return factorial(number - 1) * number;
}

int main(int argc, char **argv)
{
	if (argc != 2)
	{
		printf("Usage: factorial int\n");
		return 1;
	}

	int number = atoi(argv[1]);
	printf("%li\n", factorial(number));
}

File: java/Numbers/factorial.java
/**
 * calculate the factorial of a positive integer
 *
 */
class factorial
{
    public static long factorial(int number)
    {
        if (number == 0)
            return 1;
        else if (number == 1)
            return 1;
        else
            return factorial(number - 1) * number;
    }

    public static void main(String[] args)
    {
        if (args.length != 1)
        {
            System.out.println("Usage: java factorial int");
            System.exit(1);
        }

        int number = Integer.valueOf(args[0]);
        System.out.println(factorial(number));
    }
}
